test/metadata-extractor/src/test/java/org/elasticsearch/extractor/features/HistoricalFeaturesMetadataExtractorTests.java

```diff
@@ -33,15 +40,30 @@ public class HistoricalFeaturesMetadataExtractorTests extends ESTestCase {
 
     public void testExtractHistoricalMetadata() throws IOException {
         HistoricalFeaturesMetadataExtractor extractor = new HistoricalFeaturesMetadataExtractor(this.getClass().getClassLoader());
-        Map<NodeFeature, Version> nodeFeatureVersionMap = extractor.extractHistoricalFeatureMetadata();
+        Map<NodeFeature, Version> nodeFeatureVersionMap = new HashMap<>();
+        Set<String> featureNamesSet = new HashSet<>();
+        extractor.extractHistoricalFeatureMetadata((historical, names) -> {
+            nodeFeatureVersionMap.putAll(historical);
+            featureNamesSet.addAll(names);
+        });
         assertThat(nodeFeatureVersionMap, not(anEmptyMap()));
+        assertThat(featureNamesSet, not(empty()));
 
         Path outputFile = temporaryFolder.newFile().toPath();
         extractor.generateMetadataFile(outputFile);
         try (XContentParser parser = JsonXContent.jsonXContent.createParser(EMPTY, Files.newInputStream(outputFile))) {
-            Map<String, String> parsedMap = parser.mapStrings();
+            Map<String, Object> parsedMap = parser.map();
+            assertThat(parsedMap, hasKey("historical_features"));
+            assertThat(parsedMap, hasKey("feature_names"));
+            @SuppressWarnings("unchecked")
+            Map<String, Object> historicalFeaturesMap = (Map<String, Object>) (parsedMap.get("historical_features"));
             for (Map.Entry<NodeFeature, Version> entry : nodeFeatureVersionMap.entrySet()) {
-                assertThat(parsedMap, hasEntry(entry.getKey().id(), entry.getValue().toString()));
+                assertThat(historicalFeaturesMap, hasEntry(entry.getKey().id(), entry.getValue().toString()));
+            }
+            @SuppressWarnings("unchecked")
+            Collection<String> featureNamesList = (Collection<String>) (parsedMap.get("feature_names"));
+            for (var entry : featureNamesList) {
+                assertThat(featureNamesList, hasItem(entry));
```

Comment:
I know I have written (probably) the same test, but I cannot figure out the purpose of this assertion :D

Reply:
I haven't touched this code, beyond addressing Simons comment on your PR ;)

Reply:
You should probably write a better/more meaningful assertion anyway :)  I honestly think this is weird/not useful

Reply:
Oh, now i'm seeing it. I didn't notice before 🙈